### PR TITLE
Remove aliases

### PR DESCRIPTION
**Pull request by glasserc**
Refs #1863. This PR removes a bunch of aliases for recipe fields that actually come from revisions. The ones removed in this PR are the easy ones that aren't really used anywhere but it's a good first step and helps clarify that these aliases are exceptions rather than the rule.

@mythmon Could I have like a 10% review? Does this seem directionally accurate? I wasn't sure how much leeway I had in terms of altering the API and in particular dropping fields from serializers. The contract-tests seem to pass (modulo #2017) but maybe they aren't extensive enough?

### CODE REVIEW

**Comment by mythmon** [comment]
> The contract-tests seem to pass (modulo #2017) but maybe they aren't extensive enough?

The contract-tests don't validate that the API serves the needs of any client that I know of, beyond the level of "the API exists". The Python unit tests that we have do a better job of expressing many client uses cases that exist.

**Comment by mythmon** [comment]
Regarding changing the serializers: I don't expect that any of these changes will affect the v3 API very much, because I don't think it uses any of the aliases. The v1 API is the one we need to worry about. Technically, we shouldn't make any changes to the v1 API, though in practice we've added fields without worry.

Specifically, the [`SignedRecipeSerializer`](https://github.com/mozilla/normandy/blob/3a6e789612d41a6fa5fd33ad23e6d8fc16fa2f6c/normandy/recipes/api/v1/serializers.py#L156-L166) (and by extension `MinimalRecipeSerializer` and `SignatureSerializer` are the ones we need to be very careful changing. That collection of serializers is what generates the data that Firefox clients read to run recipes (both for legacy clients accessing the API directly and modern Remote Settings-based clients).

I don't think any of the fields you listed here affect those serializers (the tests would have broken if they did). I also don't think anyone is still using the v1 APIs except the use cases I outlined above. Once you get to the point where you're removing aliases for fields that the minimal serializer is using, you'll have to do something to keep the serializer compatible.

In short: I think this is a good direction, though it will likely get harder.

**Comment by mythmon** [comment]
@glasserc Do you plan to do anymore work on this PR?

**Comment by mythmon** [comment]
In the interest of cleaning up old PRs, I'd like to close this PR. It seems like these changes are reasonable changes that make the code better, so I'm going to merge them. We definitely have more work to do in this direction in the future.

bors merge
